### PR TITLE
added pagination to feed for #40

### PR DESCRIPTION
**Pull request by eekbatani**
I paginated the blog feed as per issue #40.

The current configuration shows three posts per page. This limit can be easily changed by changing the value of the "limitPerPage" variable at the top of pagination.js 

Even though our current blog posts are static placeholders, the pagination logic is ready to handle a dynamic feed without the need of any additional work.

### CODE REVIEW

**Comment by humphd** [comment]
@eekbatani this is failing eslint.  Please fix all the issues you see when you run `npm run eslint`

File: .eslintrc.js [hunk 1]
```diff
@@ -3,7 +3,12 @@ module.exports = {
   "env": {
     "jest": true,
     "browser": true,
+    "jquery": true
   },
+  "plugins": [
+    "dollar-sign",
```
Comment (humphd) [inline .eslintrc.js]: I'm not familiar with this plugin.  Why have you added it, and what does it offer over the `jquery` plugin you use below?

File: pagination.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/*eslint-env jquery*/
```
Comment (humphd) [inline pagination.js]: This shouldn't be necessary if you define it in the `env` in `.eslintrc.js`, which you do.  Or am I missing something?

**Comment by humphd** [comment]
You also need to rebase this and update re-run `npm install` to fix `package-lock.json`

File: pagination.js (hunk 2 above)
Comment (eekbatani) [reply pagination.js]: My apologies, you're I was having some issues with the command earlier on my end because my NodeJs was out of date. I added those lines experimentally , I'll proceed to push a cleaner version.

**Comment by cindyledev** [comment]
What's happening with that index.html file? There are so many lines being removed and added because of incorrect indentation.

**Comment by humphd** [comment]
One thing to note: in #96 we're going to start using [Prettier](https://prettier.io/), which will make this problem go away.

Having said that, @cindyledev makes a great point: we should only touch lines associated with a fix, so we don't mess up git history for the remaining code.

**Comment by eekbatani** [comment]
> What's happening with that index.html file? There are so many lines being removed and added because of incorrect indentation.

I had to add more sample blog posts to demonstrate the pagination functionality. Each post is a few paragraphs therefore many lines look like their being added.

File: .eslintrc.js [hunk 3]
```diff
@@ -27,5 +25,14 @@ module.exports = {
      * https://eslint.org/docs/rules/func-names
      */
     "func-names": "off",
+
+    "max-len": ["error", 256, 2, {
```
Comment (humphd) [inline .eslintrc.js]: Why alter these rules?  Move your comments above the line they affect vs. at the end.  Lines this long are not good.

File: .eslintrc.js [hunk 4]
```diff
@@ -27,5 +25,14 @@ module.exports = {
...
     "func-names": "off",
+
+    "max-len": ["error", 256, 2, {
+            ignoreUrls: true,
+            ignoreComments: false,
+            ignoreRegExpLiterals: true,
+            ignoreStrings: false,
+            ignoreTemplateLiterals: false,
+          }],
+    'no-plusplus': 'off'
```
Comment (humphd) [inline .eslintrc.js]: Why this change?

Comment (eekbatani) [reply .eslintrc.js]: The "no-plusplus" rule by default disables unary operators.  The unary operator is used several times in the pagination logic, and is a more efficient way to write the code as it reduces the number of lines needed. I didn't see why it should be omitted, so I set it to 'off'.

As for the rest, I will refactor the code to abide by the default rules and remove those conditions.

File: .eslintrc.js (hunk 3 above)
Comment (eekbatani) [reply .eslintrc.js]: Good point, I'll focus on changing the code instead of the rules.

File: .eslintrc.js (hunk 4 above)
Comment (humphd) [reply .eslintrc.js]: Often these things are disabled due to bugs that don't seem obvious, but people find through experience.  You can read a bit about why they disabled it in https://github.com/airbnb/javascript/issues/540.

In modern JS, using for-loops like you're doing with a lot of `i++` isn't really necessary anymore.  It's more bug prone than using for-of, `forEach`, `map`, etc.  I'd rewrite your code to not need to use an index counter at all.  That's the point of this rule.

Comment (eekbatani) [reply .eslintrc.js]: Okay fair enough I'll make the changes.

File: index.html [hunk 5]
```diff
@@ -1,255 +1,455 @@
...
-
-    <!-- Font Awesome -->
-    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css">
-    <!-- Bootstrap core CSS -->
-    <link href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/css/bootstrap.min.css" rel="stylesheet">
-    <!-- Material Design Bootstrap -->
-    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.8.10/css/mdb.min.css" rel="stylesheet">
-
-
-</head>
```
Comment (cindyledev) [inline index.html]: This doesn't needed to be removed. I see that you added the exact same lines after so it won't effect the results but this code was originally added by two other contributors. Let's not alter the git history.

Comment (eekbatani) [reply index.html]: I did not remove or move those lines (4 to 22) that you are referring to. They are still present. Only their alignment has been corrected when I made my own additions to the file as per standard HTML conventions.  As you said, this has no effect on the actual output, nor has it changed any of the work of the original contributors.
image

**Comment by humphd** [comment]
Agree, we can allow syntax changes we won't later, since we're about to reformat all the code in #139 soon.

**Comment by humphd** [comment]
A side note: one thing that this PR does that I *don't* want you to do is `Merge branch 'master' into addPagination`.  Use rebase instead when you need to update.

Looks like this has enough reviews to land, so if you're done, go ahead and merge it and we'll iterate in follow-ups.
